WPXMLRPCTest/Tests/WPXMLRPCDataCleanerTest.m

```diff
@@ -11,20 +11,61 @@ - (void)testClosingTagRepaired
 {
     WPXMLRPCDataCleaner *cleaner = [[WPXMLRPCDataCleaner alloc] init];
 
+    // Test a whole response
     NSString *completeResponse = @"<methodResponse><params><param><value><string>Hello!</string></value></param></params></methodResponse>";
-    NSString *truncatedResponse = [completeResponse substringWithRange:NSMakeRange(0, [completeResponse length] - 20)];
-    NSString *cleanedString = [cleaner cleanClosingTagIfNeeded:truncatedResponse];
-    STAssertEqualObjects(truncatedResponse, cleanedString, @"Should not try to fix a response that is too damaged.");
+    NSString *cleanedString = [cleaner cleanClosingTagIfNeeded:completeResponse lengthOfCharactersPrecedingPreamble:0];
+    STAssertEqualObjects(completeResponse, cleanedString, @"A good reponse should just be returned intact.");
 
-    completeResponse = @"<methodResponse><params><param><value><string>Hello!</string></value></param></params></methodResponse>";
+    // Test a whole response but some info was cleaned from before the preamble
```

Comment:
First off, very nice work with the amount of unit tests provided.

However, it seems to me like it could be a good idea to break these up into different tests.  It would make the code slightly more readable and the tests easier to maintain.